yinstruments/PDU_main.py

```diff
@@ -0,0 +1,236 @@
+import telnetlib
+import time
+import sys
+import os
+import re
+import pysnmp as snmp
+from abc import ABC, abstractmethod
+import subprocess
+
+
+class PDU_Port:
+    # generic class structure for any PDU systems port
+
+    def __init__(self, ip_addr, ip_port) -> None:
+        return 0
+
+    def __str__(self):
+        return 0
+
+    def reboot(self):
+        return 0
+
+    def on(self):
+        return 0
+
+    def off(self):
+        return 0
```

Comment:
These functions should be marked
`@abstractmethod`

instead of `return 0` have `raise NotImplementedError`